crates/core_arch/src/arm/mod.rs

```diff
@@ -14,24 +14,16 @@ pub use self::armclang::*;
 mod v6;
 pub use self::v6::*;
 
-#[cfg(any(target_arch = "aarch64", target_feature = "v7"))]
 mod v7;
-#[cfg(any(target_arch = "aarch64", target_feature = "v7"))]
 pub use self::v7::*;
 
-#[cfg(any(target_arch = "aarch64", target_feature = "v7", doc))]
 mod neon;
-#[cfg(any(target_arch = "aarch64", target_feature = "v7", doc))]
 pub use self::neon::*;
 
-#[cfg(any(target_arch = "aarch64", target_feature = "v7"))]
 mod crc;
-#[cfg(any(target_arch = "aarch64", target_feature = "v7"))]
 pub use self::crc::*;
 
-#[cfg(any(target_arch = "aarch64", target_feature = "v7"))]
 mod crypto;
-#[cfg(any(target_arch = "aarch64", target_feature = "v7"))]
```

Comment:
The latest commit is mostly to raise some discussion maybe. I removed a bunch of cfg and target_feature(enable = "v7") bounds that don't seem necessary to me.

Reply:
Alright so it seems that runtime detection for the ARM version isn't a thing anyway, so the former doesn't quite apply here. Though that now begs the additional question, how I even write tests for these then? The tests seem to use runtime detection (or do they?)

Reply:
The `is_arm_feature_detected` macro in `std_detect` needs to be updated to add the target features defined in https://github.com/rust-lang/rust/blob/36f1f04f18b89ba4a999bcfd6584663fd6fc1c5d/compiler/rustc_codegen_ssa/src/target_features.rs#L12.